src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -194,19 +194,14 @@ trait ClusterComponent extends LeoComponent {
     // find* and get* methods do query the INSTANCE table
 
     def getActiveClusterByName(project: GoogleProject, name: ClusterName): DBIO[Option[Cluster]] = {
-      fullClusterQuery
-        .filter { _._1.googleProject === project.value }
-        .filter { _._1.clusterName === name.value }
-        .filter { _._1.destroyedDate === Timestamp.from(dummyDate) }
+      fullClusterQueryByUniqueKey(project, name, Some(dummyDate))
         .result map { recs =>
           unmarshalFullCluster(recs).headOption
         }
     }
 
     def getDeletingClusterByName(project: GoogleProject, name: ClusterName): DBIO[Option[Cluster]] = {
-      fullClusterQuery
-        .filter { _._1.googleProject === project.value }
-        .filter { _._1.clusterName === name.value }
+      fullClusterQueryByUniqueKey(project, name, Some(dummyDate))
         .filter { _._1.status === ClusterStatus.Deleting.toString }
```

Comment:
this does a filter after the fact....but the data has already been pared down so it should be fine

Reply:
(this also wasn't the target of this perf improvement but it should still benefit from the change)